src/shogun/util/factory.h

```diff
@@ -337,5 +334,110 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+	} // namespace details
+	template <typename TypeName, typename... Args>
+	std::shared_ptr<TypeName> create(Args... args)
```

Comment:
What about
```suggestion
#ifdef SWIG
	std::shared_ptr<TypeName> create(Args... args)
#else
	std::shared_ptr<TypeName> create(Args&&... args)
#endif
```

Reply:
i added `ifdef SWIG`, but it didn't compile

Reply:
What’s the error? Compiler or swig?

Reply:
the error comes from swig, it seems like that after added `ifdef SWIG`, SWIG still use the `create(Args&&... args)`

Reply:
`#ifndef` for the win

Reply:
Like here?
https://github.com/shogun-toolbox/shogun/blob/1c235d83bc609080a9e62ecb1fc8580a0b7a6388/src/shogun/base/SGObject.h#L251-254

Reply:
yes, like this https://pastebin.com/qSKLfGiL, i don't know why it don't work

Reply:
OK, I suspect that SWIG is not parsing the template and function signature separately. Try to write the template with the function signature, so you’d have template <typename... Args> twice